### PR TITLE
Server-side addresses error handling

### PR DESCRIPTION
**Pull request by adbshiau**

### CODE REVIEW

File: server/src/controllers/appointments.controller.js [hunk 1]
```diff
@@ -34,7 +28,19 @@ export async function newAppointment(req, res, next) {
...
     const { earlyTimeHour, lateTimeHour, address, email, ...rest } = req.body;
-    const coords = await geocodeAddress(address);
+    const geocodeData = await geocodeAddress(address);
+
+    if (geocodeData.status === "ZERO_RESULTS") {
+      res.status(404);
+      return next({ message: "The address submitted is not valid." });
+    }
+
+    if (!geocodeData.results.length) {
```
Comment (DalySoftware) [inline server/src/controllers/appointments.controller.js:38]:
```suggestion
    if (!geocodeData.results) {
```
FYI this works I think

**Review by DalySoftware — DISMISSED** [review]

File: server/src/controllers/appointments.controller.js [hunk 2]
```diff
@@ -14,13 +14,7 @@ export async function geocodeAddress(address) {
   const resp = await fetch(geocodeUrl);
   const data = await resp.json();
   console.log(data);
```
Comment (tdkent) [inline server/src/controllers/appointments.controller.js:16]: This console log should be removed

File: server/src/controllers/appointments.controller.js (hunk 1 above)
Comment (tdkent) [reply server/src/controllers/appointments.controller.js:38]: `results` will be an empty array so it won't be falsy

File: server/src/controllers/appointments.controller.js [hunk 3]
```diff
@@ -45,7 +51,8 @@ export async function newAppointment(req, res, next) {
         preferredEarlyTime: earlyTimeHour,
         preferredLateTime: lateTimeHour,
       },
-      location: { ...coords, address },
+      location: geocodeData.results[0].geometry.location,
```
Comment (tdkent) [inline server/src/controllers/appointments.controller.js:54]: `location` is a subdocument so the geocoded data and the address need to be spread into a new object

**Review by tdkent — CHANGES_REQUESTED** [review]

File: server/src/controllers/appointments.controller.js (hunk 3 above)
Comment (adbshiau) [reply server/src/controllers/appointments.controller.js:54]: Do you mean like this?

location: {
        address: address,
        lat: geocodeData.results[0].geometry.location.lat,
        long: geocodeData.results[0].geometry.location.lat,
      },

Comment (tdkent) [reply server/src/controllers/appointments.controller.js:54]: That should work

**Review by tdkent — APPROVED** [review]

**Comment by tdkent** [comment]
I reordered the checkAddress and geocoding functions since its slightly better to ping the Google API once we've determined if the request is valid

**Pull request merged**